[DO NOT MERGE] Adding redesigned card pattern. closes #1455

webinista opened the pull request:
DO NOT MERGE - we're going to test this out internally via the link before rolling out.
- Updates _education/index.html with the new pattern
- Moves "GI Bill Comparison Tool" link to the GI Bill card
- Adds _va-navigation-cards.scss partial
- Adds JavaScript snippet that powers the navigation pattern
- Added width, height attributes to cards-\* SVG images to constrain their icon size

webinista committed:
Adding redesigned card pattern. closes #1455

- Updates _education/index.html with the new pattern
- Moves "GI Bill Comparison Tool" link to the GI Bill card
- Adds _va-navigation-cards.scss partial
- Adds JavaScript snippet that powers the navigation pattern
- Added width, height attributes to cards-* SVG images to constrain their icon size

webinista committed:
Adding redesigned card pattern. closes #1455

- Updates _education/index.html with the new pattern
- Moves "GI Bill Comparison Tool" link to the GI Bill card
- Adds _va-navigation-cards.scss partial
- Adds JavaScript snippet that powers the navigation pattern
- Added width, height attributes to cards-* SVG images to constrain their icon size

webinista committed:
Restoring hover state box shadow.

webinista committed:
Restoring hover state box shadow after merging remote branch.

Commit:
Merge branch 'master' into card-redesign

emilyville commented:
it looks good, but for some reason the interaction isn't working for me locally. I will look again tomorrow!

emilyville commented:
I'm still having issues. Maybe someone else can pull this down and then I'll know if it's just my build? Seems quite possible.

webinista committed:
Merge branch 'master' into card-redesign

akainic commented on _education/index.html:
```diff
@@ -17,49 +17,50 @@
 
     <div class="row">
-      <div class="small-12 columns">
+      <div class="small-12">
```
Why do you want to delete `columns`?

webinista replied:
Columns adds padding on either side, but that isn't really necessary.

akainic commented on assets/js/vendor/cards-expandable.js:
```diff
@@ -0,0 +1,12 @@
+    		e.preventDefault();
+    	}
+	});
+});
```
Did this js come from somewhere? If so, I think we should include a comment above describing where it came from. If not, we should probably move it out of the vendor directory.

akainic commented on assets/js/vendor/cards-expandable.js:
```diff
@@ -0,0 +1,12 @@
+// JS for expandable cards (.va-card--expands)
+$(document).ready(function() {
+	$('.va-card-body').on('click', function(e){
+    	if(e.target.nodeName == 'DIV') {
```
I think it's best to use `===` instead of `==`.

akainic replied:
This is a bit of a weird mix between jQuery and regular JavaScript. I see why you wanted to use `e.target` instead of `this` because you want to check if what was clicked on was the whole card or just the header, but it reads a little weird. I see why you would want to use jQuery, because it does make some of the handling of checking for and changing classes easier. I wonder if it's better to use `$(e.target).is('div')` instead? I was also thinking that you could use `this` when checking the card itself, but maybe best to make your 2 checks here consistent. Not sure what the best solution is, just throwing out ideas.

webinista replied:
Using `is('div')` makes sense. The problem with using `this` is that the click target might be a child of `.va-card-body`. But checking the `currentTarget` might be better here.

akainic commented on assets/js/vendor/cards-expandable.js:
```diff
@@ -0,0 +1,12 @@
+    		$(e.currentTarget).parent().toggleClass('va-card--hasdrawer--open');
+    	}
+    	// If the drawer is open, disable links in the card's body
+    	if(e.target.nodeName == 'A' && $(e.currentTarget).parent().hasClass('va-card--hasdrawer--open')) {
```
Is this to disable the `h4` in the card when the drawer is open? If so, why doesn't the drawer just cover the whole card if we don't want the header to be clickable anyway?

webinista replied:
@akainic I agree with that, tbh, but that's not what the original design was. 

@gnakm Can you weigh in here? It's an easy change to make, and would simplify the JavaScript a bit.

akainic commented:
@emilyville I think the interaction is working for me locally!

webinista committed:
Merge branch 'master' into card-redesign

emilyville commented:
I think this looks good. Let's use the deployment URL to share and gather feedback before rolling out. Assigning to @gnakm for this!

https://vets-website-pr-1478.herokuapp.com/

webinista committed:
Merge branch 'master' into card-redesign

webinista committed:
Merge branch 'master' into card-redesign

webinista committed:
Merge branch 'master' into card-redesign

webinista commented:
Assigning this back to myself for a bit while I address some of Anne's comments.

webinista committed:
Switching e.target.nodeName to $(e.target).is()

Also renamed:    assets/js/vendor/cards-expandable.js -> assets/js/cards-expandable.js

webinista committed:
Updating path to cards-expandable.js

Commit:
Merge branch 'master' into card-redesign

webinista committed:
Merge branch 'master' into card-redesign

webinista committed:
Merge branch 'master' into card-redesign

webinista committed:
Merge branch 'master' into card-redesign

gnakm commented:
Closing this because we got feedback from vets.gov team. Moving forward design team will consider when to use blue cards and solve the sublink issue.
